In [8]:
import pandas as pd

dados = pd.read_csv('DENGBR25.csv', sep=',')

/tmp/ipython-input-31766338.py:3: DtypeWarning: Columns (44,45,46,54,74,101) have mixed types. Specify dtype option on import or set low_memory=False.
  dados = pd.read_csv('DENGBR25.csv', sep=',')


### Verificando que as colunas do Warning ao conectar com o dataset são do tipo "object", para logo verificar possível tratamento.

In [9]:
dados.iloc[:, [44, 45, 46, 54, 101]].dtypes



,0
DT_CHIK_S1,object
DT_CHIK_S2,object
DT_PRNT,object
DT_VIRAL,object
DT_GRAV,object


In [10]:
dados.iloc[:, [44,45,46,54,101]].head(5)

,DT_CHIK_S1,DT_CHIK_S2,DT_PRNT,DT_VIRAL,DT_GRAV
0,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN
2,NaN,NaN,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN
4,NaN,NaN,NaN,NaN,NaN


In [11]:
duplicados = dados[dados.duplicated()]

qtd_duplicados = dados.duplicated().sum()
print(f"Quantidade de registros duplicados: {qtd_duplicados}")

Quantidade de registros duplicados: 649




### Quantidade de 649 registros duplicados removidos.

In [12]:
dados = dados.drop_duplicates()

In [13]:
dados.isna().sum()

,0
TP_NOT,0
ID_AGRAVO,0
DT_NOTIFIC,0
SEM_NOT,0
NU_ANO,0
...,...
NDUPLIC_N,1617083
DT_DIGITA,420
CS_FLXRET,1
FLXRECEBI,1621788


### Quantidade de 1621789 registros e 121 colunas.

In [14]:
dados.shape

(1621789, 121)

### Inferência sobre os dados - Avaliando as métricas

- Quantidade de registros duplicados 649, foram removidos.
- Muitos campos com registros NaN, esses campos serão tratados um a um de acordo com sua particularidade no dicionario de dados. URL: https://s3.sa-east-1.amazonaws.com/ckan.saude.gov.br/SINAN/Dengue/dic_dados_dengue.pdf


- Os campos com registros NaN foram:

  ID_REGIONA -> 49059 registros -> coluna 8
  
  ID_UNIDADE -> 161 registros -> coluna 9
  
  ANO_NASC -> 4626 registros -> coluna 12
  
  CS_SEXO -> 1 registro -> coluna 14
  
  CS_GESTANT -> 176 registros -> coluna 15
  
  CS_RACA -> 2 registros -> coluna 16
  
  CS_ESCOL_N -> 309300 registros -> coluna 17
  
  ID_MN_RESI -> 7 registros -> coluna 19
  
  ID_RG_RESI -> 49425 registros -> coluna 20
  
  DT_INVEST -> 66981 registros -> coluna 22
  
  ID_OCUPA_N -> 1181590 registros -> coluna 23
  
  FEBRE -> 67003 registros -> coluna 24
  
  MIALGIA -> 67003 registros -> coluna 25
  
  CEFALEIA -> 67003 registros -> coluna 26
  
  EXANTEMA -> 67003 registros -> coluna 27
  
  VOMITO -> 67003 registros -> coluna 28
  
  NAUSEA -> 67003 registros -> coluna 29
  
  DOR_COSTAS -> 67003 registros -> coluna 30
  
  CONJUNTVIT -> 67003 registros -> coluna 31
  
  ARTRITE -> 67003 registros -> coluna 32
  
  ARTRALGIA -> 67003 registros -> coluna 33
  
  PETEQUIA_N -> 67003 registros -> coluna 34
  
  LEUCOPENIA -> 67003 registros -> coluna 35
  
  LACO -> 67003 registros -> coluna 36
  
  DOR_RETRO -> 67003 registros -> coluna 37
  
  DIABETES -> 67003 registros -> coluna 38
  
  HEMATOLOG -> 67003 registros -> coluna 39
  
  HEPATOPAT -> 67003 registros -> coluna 40
  
  RENAL -> 67003 registros -> coluna 41
  
  HIPERTENSA -> 67003 registros -> coluna 42
  
  ACIDO_PEPT -> 67003 registros -> coluna 43
  
  AUTO_IMUNE -> 67003 registros -> coluna 44
  
  DT_CHIK_S1 -> 1620642 registros -> coluna 45
  
  DT_CHIK_S2 -> 1621784 registros-> coluna 46
  
  DT_PRNT -> 1621465 registros -> coluna 47
  
  RES_CHIKS1 -> 1618051 registros -> coluna 48
  
  RES_CHIKS2 -> 1618432 registros -> coluna 49
  
  RESUL_PRNT -> 1618248 registros -> coluna 50
  
  DT_SORO -> 1358302 registros -> coluna 51
  
  RESUL_SORO -> 710248 registros -> coluna 52
  
  DT_NS1 -> 1092459 registros -> coluna 53
  
  RESUL_NS1 -> 581091 registros -> coluna 54
  
  DT_VIRAL -> 1613279 registros -> coluna 55
  
  RESUL_VI_N -> 873813 registros -> coluna 56
  
  DT_PCR -> 1558324 registros -> coluna 57
  
  RESUL_PCR_ -> 838729 registros -> coluna 58
  
  SOROTIPO -> 1574636 registros -> coluna 59
  
  HISTOPA_N -> 946909 registros -> coluna 60
  
  IMUNOH_N -> 946741 registros -> coluna 61
  
  HOSPITALIZ -> 394383 registros -> coluna 62
  
  DT_INTERNA -> 1556825 registros -> coluna 63
  
  * COUFINF -> 512143 registros -> coluna 64
  
  MUNICIPIO -> 1557144 registros -> coluna 65
  
  TPAUTOCTO -> 468113 registros -> coluna 66
  
  * COUFINF -> 512143 registros -> coluna 67
  
  COPAISINF -> 152013 registros -> coluna 68
  
  COMUNINF -> 152808 registros -> coluna 69
  
  CLASSI_FIN -> 15825 registros -> coluna 70
  
  CRITERIO -> 89050 registros -> coluna 71
  
  DOENCA_TRA -> 324933 registros -> coluna 72
  
  CLINIC_CHIK -> nao achou a coluna -> coluna  73
  
  EVOLUCAO -> 121045 registros -> coluna 74
  
  DT_OBITO -> 324479 registros -> coluna 75
  
  DT_ENCERRA -> 18993 registros -> coluna 76
  
  ALRM_HIPOT -> 317834 registros -> coluna 77
  
  ALRM_PLAQ -> 317803 registros -> coluna 78
  
  ALRM_VOM -> 317827 registros -> coluna 79
  
  ALRM_SANG -> 317825 registros -> coluna 80
  
  ALRM_HEMAT -> 317850 registros -> coluna 81
  
  ALRM_ABDOM -> 317780 registros -> coluna 82
  
  ALRM_LETAR -> 317851 registros -> coluna 83
  
  ALRM_HEPAT -> 317856 registros -> coluna 84
  
  ALRM_LIQ -> 317859 registros -> coluna 85
  
  DT_ALRM 317874 registros -> coluna 86
  
  GRAV_PULSO -> 324430 registros -> coluna 87
  
  GRAV_CONV -> 324434 registros -> coluna 88
  
  GRAV_ENCH -> 324435 registros -> coluna 89
  
  GRAV_INSUF -> 324437 registros -> coluna 90
  
  GRAV_TAQUI -> 324437 registros -> coluna 91
  
  GRAV_EXTRE -> 324436 registros -> coluna 92
  
  GRAV_HIPOT -> 324437 registros -> coluna 93
  
  GRAV_HEMAT -> 324436 registros -> coluna 94
  
  GRAV_MELEN -> 324436 registros -> coluna 95
  
  GRAV_METRO -> 324438 registros -> coluna 96
  
  GRAV_SANG -> 324437 registros -> coluna 97
  
  GRAV_AST -> 324439 registros -> coluna 98
  
  GRAV_MIOC -> 324439 registros -> coluna 99
  
  GRAV_CONSC -> 324437 registros -> coluna 100
  
  GRAV_ORGAO -> 324439 registros -> coluna 101
  
  DT_GRAV -> 324441 registros -> coluna 102
  
  MANI_HEMOR -> 324933 registros -> coluna 103
  
  EPISTAXE -> 324933 registros -> coluna 104
  
  GENGIVO -> 324933 registros -> coluna 105
  
  METRO -> 324933 registros -> coluna 106
  
  PETEQUIAS -> 324933 registros -> coluna 107
  
  HEMATURA -> 1621789 registros -> coluna 108 -> todos os campos estao nulos
  
  SANGRAM -> 1621789 registros -> coluna 109 -> todos os campos estao nulos
  
  LACO_N -> 1621789 registros -> coluna 110 -> todos os campos estao nulos
  
  PLASMATICO -> 1621789 registros -> coluna 111 -> todos os campos estao nulos
  
  EVIDENCIA -> 1621789 registros -> coluna 112 -> todos os campos estao nulos
  
  PLAQ_MENOR -> 1621789 registros -> coluna 113 -> todos os campos estao nulos
  
  CON_FHD -> 1621789 registros -> coluna 114 -> todos os campos estao nulos
  
  COMPLICA -> 1621789 registros -> coluna 115 -> todos os campos estao nulos
  
  TP_SISTEMA -> 18456 registros -> coluna 116
  
  NDUPLIC_N -> 1617083 registros -> coluna 117
  
  DT_DIGITA -> 420 registros -> coluna 118
  
  CS_FLXRET -> 1 registros -> coluna 119
  
  FLXRECEBI -> 1621788 registros -> coluna 120
  
  MIGRADO_W -> 1621789 registros-> coluna 121
  

In [ ]:
dados['NDUPLIC_N'][dados['NDUPLIC_N'].notna()]


,NDUPLIC_N
1306,1.0
6839,1.0
6840,1.0
7342,1.0
16410,1.0
...,...
1619708,1.0
1619795,1.0
1619960,1.0
1621363,1.0


In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18767 entries, 0 to 18766
Columns: 121 entries, TP_NOT to MIGRADO_W
dtypes: float64(91), int64(12), object(18)
memory usage: 17.3+ MB


In [ ]:
dados.head()

,TP_NOT,ID_AGRAVO,DT_NOTIFIC,SEM_NOT,NU_ANO,SG_UF_NOT,ID_MUNICIP,ID_REGIONA,ID_UNIDADE,DT_SIN_PRI,...,EVIDENCIA,PLAQ_MENOR,CON_FHD,COMPLICA,TP_SISTEMA,NDUPLIC_N,DT_DIGITA,CS_FLXRET,FLXRECEBI,MIGRADO_W
0,2,A90,2025-11-26,202548,2025,15,150555,1495.0,2615975.0,2025-11-18,...,NaN,NaN,NaN,NaN,2.0,NaN,2025-11-28,0.0,NaN,NaN
1,2,A90,2025-11-26,202548,2025,15,150555,1495.0,2615975.0,2025-11-26,...,NaN,NaN,NaN,NaN,2.0,NaN,2025-11-28,0.0,NaN,NaN
2,2,A90,2025-12-08,202550,2025,15,150555,1495.0,2615975.0,2025-12-07,...,NaN,NaN,NaN,NaN,2.0,NaN,2025-12-12,0.0,NaN,NaN
3,2,A90,2025-12-08,202550,2025,15,150555,1495.0,2615975.0,2025-12-07,...,NaN,NaN,NaN,NaN,2.0,NaN,2025-12-12,0.0,NaN,NaN
4,2,A90,2025-12-08,202550,2025,15,150555,1495.0,2615975.0,2025-12-07,...,NaN,NaN,NaN,NaN,2.0,NaN,2025-12-12,0.0,NaN,NaN
